### PR TITLE
modify code

### PR DESCRIPTION
**Pull request by Galaxy1458**
### PR Category
Other

### Type of Change
Refactor 

### Description
modify some code to support muti_backend

### Progress

- [x] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### CODE REVIEW

File: src/flag_gems/ops/upsample_bicubic2d_aa.py [hunk 1]
```diff
@@ -378,7 +366,10 @@ def upsample_bicubic2d_aa_kernel(
 
 
 # upsample and downsample
-@triton.autotune(configs=configs(), key=["N", "C", "OH", "OW"])
+@triton.autotune(
+    configs=runtime.get_triton_config("upsample_bicubic2d_aa"),
+    key=["N", "C", "OH", "OW"],
+)
 @triton.jit
 def general_interpolate_bicubic2d_aa_kernel(
```
Comment (StrongSpoon) [inline src/flag_gems/ops/upsample_bicubic2d_aa.py:374]: suggest adding device guard for it incidentally.

File: src/flag_gems/ops/upsample_nearest2d.py [hunk 2]
```diff
@@ -5,21 +5,16 @@
...
-    return [
-        triton.Config({"BLOCK_SIZE": bs}, num_warps=wp) for bs in block for wp in warps
-    ]
-
-
-@triton.autotune(configs=configs(), key=["N", "C", "OH", "OW"])
+@triton.autotune(
+    configs=runtime.get_triton_config("upsample_nearest2d"), key=["N", "C", "OH", "OW"]
+)
 @triton.heuristics(
```
Comment (StrongSpoon) [inline src/flag_gems/ops/upsample_nearest2d.py:18]: ditto

File: src/flag_gems/runtime/backend/__init__.py [hunk 3]
```diff
@@ -115,11 +115,7 @@ def get_vendor_infos():
 def get_curent_device_extend_op(vendor_name=None):
     global vendor_module
     get_vendor_module(vendor_name)
-    tuples = vendor_module.get_register_op_config()
-    configs = {}
-    for item in tuples:
-        configs[item[0]] = item
-    return configs
+    return vendor_module.get_register_op_config()
```
Comment (StrongSpoon) [inline src/flag_gems/runtime/backend/__init__.py:118]: what about returning it as a list to allow concatenate?

File: src/flag_gems/ops/upsample_bicubic2d_aa.py (hunk 1 above)
Comment (Galaxy1458) [reply src/flag_gems/ops/upsample_bicubic2d_aa.py:374]: Done

File: src/flag_gems/ops/upsample_nearest2d.py (hunk 2 above)
Comment (Galaxy1458) [reply src/flag_gems/ops/upsample_nearest2d.py:18]: Done

File: src/flag_gems/runtime/backend/__init__.py (hunk 3 above)
Comment (Galaxy1458) [reply src/flag_gems/runtime/backend/__init__.py:118]: Done

**Review by StrongSpoon — DISMISSED** [review]
lgtm

**Review by StrongSpoon — APPROVED** [review]
lgtm

**Pull request merged**